examples/http2/index.js

```diff
@@ -0,0 +1,40 @@
+var isHttp2Supported = require('../../lib/utils').isHttp2Supported;
+if (isHttp2Supported) {
+  var fs = require('fs');
+  var path = require('path');
+  var express = require('../../');
+  var http2 = require('http2');
+  var keys = path.join(__dirname, 'keys');
+  var app = express();
+  var style = fs.readFileSync('./static/style.css', 'utf8');
+  function pushStyle(res) {
+    res.createPushResponse({
+      ':path': '/style.css',
+      ':status': 200,
+      'content-type': 'text/css'
+    }, function(err, newResponse) {
+      newResponse.setHeader('content-type', 'text/css');
+      newResponse.end(style);
+    });
+  }
+
+  app.get('/', function(req, res, next){
+    pushStyle(res);
+    next();
+  });
+  app.get('/index.html', function(req, res, next) {

```

Comment:
I see you are trying to show an example of pushing thr dtyle sheet for html. You missed /index works as well. Thr static middleware has a callback you can use to see the file it is about to send, which would be a better example instead of trying to enumerate them as routes.

Reply:
I didn't know about the static callback :) 
How do I set it up? did you mean the setHeaders callback?

Reply:
Sorry, on holiday and wasn't easy to lookup :) so the setHeader could be used, but seems to really be an abuse of it. Looking into it, what I was thinking was the "file" event that is in send. So I wonder if as part of getting serve-static/send/deps working with http2, another callback should be added in order to make pushing file assets very straight forward.

Reply:
Adding support for the new http2 push is something I was wondering how to do, because I think the best way express could solve it is to make it go through another route when you hit one route you know would require the other. i.e. when requiring index.html send it also to style.css (assuming no static is used), but I have no clue where to begin ;) 

How do  I use that file event from send? I thought it was pretty deep inside.